tests/integration/SparqlLuaQueryRunnerIntegrationTest.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace ProfessionalWiki\SPARQL\Tests\Integration;
+
+use MediaWiki\MediaWikiServices;
+use PHPUnit\Framework\TestCase;
+use ProfessionalWiki\SPARQL\SparqlLuaQueryRunner;
+
+/**
+ * @covers SparqlLuaQueryRunner
+ */
+class SparqlLuaQueryRunnerIntegrationTest extends TestCase {
+
+	private SparqlLuaQueryRunner $queryRunner;
+
+	protected function setUp(): void {
+		$sparqlEndpoint = MediaWikiServices::getInstance()->getMainConfig()->get( 'SPARQLEndpoint' );
+
+		if ( !is_string( $sparqlEndpoint ) ) {
+			$this->markTestSkipped( 'Missing required config variable SPARQLEndpoint' );
+		}
+
+		$requestFactory = MediaWikiServices::getInstance()->getHttpRequestFactory();
+		$this->queryRunner = new SparqlLuaQueryRunner( $requestFactory, $sparqlEndpoint );
+	}
+
+	public function testHappyPath(): void {
+		$this->assertInstanceOf( SparqlLuaQueryRunner::class, $this->queryRunner );
+	}
+
+	public function testRunQueryWhenSparqlQueryIsEmpty(): void {
+		$sparqlQuery = '';
+		$result = $this->queryRunner->runQuery( $sparqlQuery );
+		$this->assertNull( $result );
+	}
+
+	public function testRunQueryReturnsRightHeadVars(): void {
+		$sparqlQuery = "SELECT ?subject ?predicate ?object WHERE { ?subject ?predicate ?object } LIMIT 1";
+		$response = $this->queryRunner->runQuery( $sparqlQuery );
+		$data = json_decode( $response, true );
+
+		$this->assertArrayHasKey( 'head', $data );
+		$this->assertArrayHasKey( 'vars', $data['head'] );
+		$this->assertSame( [ 'subject', 'predicate', 'object' ], $data['head']['vars'] );
+	}
+
+	public function testRunQueryReturnsRightNoOfResults(): void {
+		$limit = 3;
+		$sparqlQuery = "SELECT ?subject ?predicate ?object WHERE { ?subject ?predicate ?object } LIMIT $limit";
+		$response = $this->queryRunner->runQuery( $sparqlQuery );
+		$responseData = json_decode( $response, true );
+
+		$this->assertArrayHasKey( 'results', $responseData );
+		$this->assertArrayHasKey( 'bindings', $responseData['results'] );
+		$this->assertCount( $limit, $responseData['results']['bindings'] );
+	}
```

Comment:
Interesting tests. Do they depend on test data being inserted into Blazegraph?

Reply:
They don't depend on that. It makes actual requests to the configured $sparqlEndpoint. If it's not configured, tests are skipped. Tests should ideally pass on all ENVs and platforms as long as we can make HTTP requests with the right endpoint.

Reply:
I will verify after our meeting by doing local setup